[CDN] Copy changes from API version 2019-04-15 into 2019-06-15-preview.

fore5fire opened the pull request:
This change merges changes that have already been approved from the latest stable `cdn.json` swagger file into the preview version (2019-04-15 > 2019-06-15-preview). It also adds new functionality into `cdnwebapplicationfirewall.json`, but breaking changes aren't a concern because this functionality is not yet released to customers.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

sushantsingh committed:
Update ActionType

Removed some unused fields from ActionType

sushantsingh committed:
Add range for priority

Commit:
Copy changes from CDN version 2019-04-15 into 2019-06-15-preview.

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-go

azuresdkci commented:
Can one of the admins verify this patch?

Commit:
Fix cdnwebapplicationfirewall.json syntax.

Commit:
Add transforms to CDN WAF MatchConditions

lsmith130 commented:
@sergey-shandar What do I need to do to get this merged?

sushantsingh commented:
@sergey-shandar Can you please help review it? We are releasing this product soon and are blocked on this swagger change to release Powershell and CLi for this product.

Commit:
Sync latest changes from CDN 2019-04-15 to 2019-06-15-preview

Commit:
Fix CDN WAF examples for updated API version 2019-06-15-preview.

sanjaiganesh commented on specification/cdn/resource-manager/Microsoft.Cdn/preview/2019-06-15-preview/examples/WafListPolicies.json:
```diff
@@ -108,7 +108,7 @@
               },
-              "cdnEndpointLinks": [
+              "endpointLinks": [
                 {
```
Please update allt he samples with other fields as well .. TransformType and ActionType

lsmith130 replied:
I've updated the examples with the appropriate transforms fields.

sanjaiganesh commented:
Adding ARMSignoff.. 
From offline discussion:  
No customers for the Preview version .. 
CDN.json is from the stable version.. 

So, no breaking changes as such.

Commit:
Add transforms examples for CDN API version
2019-06-15-preview.

Merge updates to CDN examples from API version 2019-04-15
to 2019-06-15-preview.

lsmith130 commented:
@sergey-shandar, Arm team has signed off - What do I need to do to get this merged?
